### PR TITLE
Update audit_rules_path_syscall template

### PR DESCRIPTION
**Pull request by shawndwells**

### CODE REVIEW

**Comment by shawndwells** [comment]
Bash remediations now included in datastream, e.g.

`````
              <fix id="audit_rules_etc_group_open" system="urn:xccdf:fix:script:sh">

# First perform the remediation of the syscall rule
# Retrieve hardware architecture of the underlying system
[ "$(getconf LONG_BIT)" = "32" ] &amp;&amp; RULE_ARCHS=("b32") || RULE_ARCHS=("b32" "b64")

for ARCH in "${RULE_ARCHS[@]}"
do
        PATTERN="-a always,exit -F arch=$ARCH -S  -F a2&amp;03 -F path={{ PATH }}.*"
        GROUP="access"
        FULL_RULE="-a always,exit -F arch=$ARCH -S  -F a2&amp;03 -F path={{ PATH }} -F auid&gt;=1000 -F auid!=unset -F key=user-modify"
        # Perform the remediation for both possible tools: 'auditctl' and 'augenrules'
<sub idref="xccdf_org.ssgproject.content_value_function_fix_audit_syscall_rule" use="legacy"/>
        fix_audit_syscall_rule "auditctl" "$PATTERN" "$GROUP" "$ARCH" "$FULL_RULE"
        fix_audit_syscall_rule "augenrules" "$PATTERN" "$GROUP" "$ARCH" "$FULL_RULE"
done

for ARCH in "${RULE_ARCHS[@]}"
do
        PATTERN="-a always,exit -F arch=$ARCH -S  -F a2&amp;03 -F path={{ PATH }}.*"
        GROUP="access"
        FULL_RULE="-a always,exit -F arch=$ARCH -S  -F a2&amp;03 -F path={{ PATH }} -F auid&gt;=1000 -F auid!=unset -F key=user-modify"
        # Perform the remediation for both possible tools: 'auditctl' and 'augenrules'
<sub idref="xccdf_org.ssgproject.content_value_function_fix_audit_syscall_rule" use="legacy"/>
        fix_audit_syscall_rule "auditctl" "$PATTERN" "$GROUP" "$ARCH" "$FULL_RULE"
        fix_audit_syscall_rule "augenrules" "$PATTERN" "$GROUP" "$ARCH" "$FULL_RULE"
done
</fix>
`````

**Comment by shawndwells** [comment]
Should be ready for review. Now includes bash and ansible remediations. Build passes locally.

**Comment by shawndwells** [comment]
Attempted remediation and receiving errors...

`````

Title   Record Events that Modify User/Group Information via open_by_handle_at syscall - /etc/group
Rule    xccdf_org.ssgproject.content_rule_audit_rules_etc_group_open_by_handle_at
Ident   CCE-80929-3
Result  error

Title   Record Events that Modify User/Group Information via open syscall - /etc/passwd
Rule    xccdf_org.ssgproject.content_rule_audit_rules_etc_passwd_open
Ident   CCE-80930-1
Result  error

Title   Record Events that Modify User/Group Information via openat syscall - /etc/group
Rule    xccdf_org.ssgproject.content_rule_audit_rules_etc_group_openat
Ident   CCE-80928-5
Result  error

Title   Record Events that Modify User/Group Information via open syscall - /etc/group
Rule    xccdf_org.ssgproject.content_rule_audit_rules_etc_group_open
Ident   CCE-80927-7
Result  error

Title   Record Events that Modify User/Group Information via openat syscall - /etc/passwd
Rule    xccdf_org.ssgproject.content_rule_audit_rules_etc_passwd_openat
Ident   CCE-80931-9
Result  error

`````

**Comment by shawndwells** [comment]
Not sure what to do here. The remediation templates assume that scap-security-guide RPM is installed and has the functions placed into `/usr/share/scap-security-guide/remediation_functions`. This makes it impossible to test remediations. The build system no longer creates RPMs so this file can't be created during development.

How do others handle this?

**Comment by redhatrises** [comment]
> Not sure what to do here. The remediation templates assume that scap-security-guide RPM is installed and has the functions placed into `/usr/share/scap-security-guide/remediation_functions`. This makes it impossible to test remediations. The build system no longer creates RPMs so this file can't be created during development.
> 
> How do others handle this?

`/usr/share/scap-security-guide/remediation_functions` hasn't been delivered in the rpm for a while now. The bash scripts use this line to add and map to the remediation functions in XCCDF. You need to properly source from `shared/bash_remediation_functions`.

File: shared/templates/template_BASH_audit_rules_path_syscall [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+. /usr/share/scap-security-guide/remediation_functions
+
+# First perform the remediation of the syscall rule
+# Retrieve hardware architecture of the underlying system
+[ "$(getconf LONG_BIT)" = "32" ] && RULE_ARCHS=("b32") || RULE_ARCHS=("b32" "b64")
+
+for ARCH in "${RULE_ARCHS[@]}"
+do
+	PATTERN="-a always,exit -F arch=$ARCH -S {{{ NAME }}} -F a2&03 -F path={{ PATH }}.*"
+	GROUP="access"
```
Comment (redhatrises) [inline shared/templates/template_BASH_audit_rules_path_syscall]: @shawndwells change the group to `modify`

File: shared/templates/template_BASH_audit_rules_path_syscall [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+	FULL_RULE="-a always,exit -F arch=$ARCH -S {{{ NAME }}} -F a2&03 -F path={{ PATH }} -F auid>={{{ auid }}} -F auid!=unset -F key=modify"
+	# Perform the remediation for both possible tools: 'auditctl' and 'augenrules'
+	fix_audit_syscall_rule "auditctl" "$PATTERN" "$GROUP" "$ARCH" "$FULL_RULE"
+	fix_audit_syscall_rule "augenrules" "$PATTERN" "$GROUP" "$ARCH" "$FULL_RULE"
+done
+
+for ARCH in "${RULE_ARCHS[@]}"
+do
+        PATTERN="-a always,exit -F arch=$ARCH -S {{{ NAME }}} -F a2&03 -F path={{ PATH }}.*"
+        GROUP="access"
```
Comment (redhatrises) [inline shared/templates/template_BASH_audit_rules_path_syscall]: @shawndwells change the group to `modify`

File: shared/templates/template_BASH_audit_rules_path_syscall (hunk 1 above)
Comment (shawndwells) [reply shared/templates/template_BASH_audit_rules_path_syscall]: Done

File: shared/templates/template_BASH_audit_rules_path_syscall (hunk 2 above)
Comment (shawndwells) [reply shared/templates/template_BASH_audit_rules_path_syscall]: Done

**Comment by shawndwells** [comment]
> > Not sure what to do here. The remediation templates assume that scap-security-guide RPM is installed and has the functions placed into `/usr/share/scap-security-guide/remediation_functions`. This makes it impossible to test remediations. The build system no longer creates RPMs so this file can't be created during development.
> > How do others handle this?
> 
> `/usr/share/scap-security-guide/remediation_functions` hasn't been delivered in the rpm for a while now. The bash scripts use this line to add and map to the remediation functions in XCCDF. You need to properly source from `shared/bash_remediation_functions`.

Ah, yes, thanks. Stepped away to take a break and came back... works now. Thanks!

**Comment by jan-cerny** [comment]
> /usr/share/scap-security-guide/remediation_functions hasn't been delivered in the rpm for a while now. The bash scripts use this line to add and map to the remediation functions in XCCDF. You need to properly source from shared/bash_remediation_functions.

This is a candidate to "Most confusing thing in ComplianceAsCode Award 2019". We need to do something with that.

File: shared/templates/template_BASH_audit_rules_path_syscall [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+for ARCH in "${RULE_ARCHS[@]}"
+do
+	PATTERN="-a always,exit -F arch=$ARCH -S {{{ SYSCALL }}} -F a2&03 -F path={{{ PATH }}}.*"
+	GROUP="modify"
+	FULL_RULE="-a always,exit -F arch=$ARCH -S {{{ SYSCALL }}} -F a2&03 -F path={{{ PATH }}} -F auid>={{{ auid }}} -F auid!=unset -F key=modify"
+	# Perform the remediation for both possible tools: 'auditctl' and 'augenrules'
+	fix_audit_syscall_rule "auditctl" "$PATTERN" "$GROUP" "$ARCH" "$FULL_RULE"
+	fix_audit_syscall_rule "augenrules" "$PATTERN" "$GROUP" "$ARCH" "$FULL_RULE"
+done
+
```
Comment (yuumasato) [inline shared/templates/template_BASH_audit_rules_path_syscall]: The `for loop` above, and below this comment seem identical to me.
Do we need both of them?

**Comment by yuumasato** [comment]
@shawndwells Do mind if take over, rebase and resolve the issues?

**Comment by redhatrises** [comment]
@yuumasato please take over and fix for @shawndwells

**Comment by shawndwells** [comment]
Please do!

-- 
Shawn Wells
Chief Security Strategist
North America Public Sector
shawn@redhat.com   |   443-534-0130

**Comment by yuumasato** [comment]
I have rebased and updated.

**Comment by matusmarhefka** [comment]
@yuumasato LGTM, thanks for the updates. I have also tested this on clean rhel8 installation and also using SSGTS on Fedora.

**Pull request merged**